docs/source/development/cdf_guide.rst

```diff
@@ -0,0 +1,686 @@
+
+Attributes come in two forms:
+
+* **Variable Attributes** - Attributes can be attached to a Variable if they only describe a particular data variable.
+* **Global Attributes** - If they are not attached to a variable, they describe the file in it's entirety.
+
+
+ISTP Compliance
+===============
+
+ISTP compliance is a standard set of attributes to use when creating CDF files.  The goal of ISTP is that others users and programs of your CDF file will know where to look for information about the data inside.  A large part of these requirements are driven by the SPDF's display tool, `CDAWeb <https://cdaweb.gsfc.nasa.gov/cdaweb/>`_.
+
+.. image:: ../_static/2d_spectrogram_1.gif
+
+
+The image above is taken from CDAWeb. Several of the attributes inside the CDF file are used to help generate these plots.
+
+To archive at the SPDF, files must be ISTP compliant.  The SPDF will reject files that do not conform to their standards.
+
+A guide to SPDF can be found here: `https://spdf.gsfc.nasa.gov/istp_guide/istp_guide.html <https://spdf.gsfc.nasa.gov/istp_guide/istp_guide.html>`_
+
+*****************************
+cdflib.xarray_to_cdf
+*****************************
+
+The Python library `cdflib <https://github.com/MAVENSDC/cdflib>`_ is a pure-python implementation of the CDF libraries.  Of particular note is the function `xarray_to_cdf <https://cdflib.readthedocs.io/en/latest/api/cdflib.xarray.xarray_to_cdf.html#cdflib.xarray.xarray_to_cdf>`_.
+
+**IMAP L0-L2 processing will be using this function to generate all CDF files**.
+
+This function takes an `xarray.Dataset <https://docs.xarray.dev/en/stable/generated/xarray.Dataset.html>`_ object as input, and will output a CDF file.  Further details about how it works are below.
+
+Xarray Introduction
+===================
+
+Xarray is a powerful python library for handling multi-dimensional data.  It has a strong connection with the netCDF file format.  The developers intended xaray to be an "in-memory" representation of a netCDF file.
+
+Perhaps the largest difference between netCDF and CDF is that netCDF has built-in methods to attach *dimensions* and *coordinates* to data.  Similarly, xarray Datasets have this capability as well.
+Full documentation about xarray Dataset objects are located here `https://docs.xarray.dev/en/stable/generated/xarray.Dataset.html <https://docs.xarray.dev/en/stable/generated/xarray.Dataset.html>`_
+
+Within a CDF file, there is no inherent way to attach coordinates to a variable.  CDF is a simpler format, and only has the concept of Attributes and Variables.  The "CDF-native" way of specifying dimensions to a variable is to use "DEPEND" attributes (see sections :ref:`ISTP compliance` and :ref:`DEPEND_i`).
+
+.. note:: the netCDF file format was created by NOAA a few years after the CDF file format to plug in some of the shortfallings of the CDF file format.  These days, the orginal reasons for the split between the two formats have largely disappeared.
+
+xarray_to_cdf
```

Comment:
cdflib.xarray_to_cdf ? Link to your code here as well if there is documentation for this method already?